newton/_src/solvers/featherstone/kernels.py

```
@@ -720,79 +588,126 @@ def compute_link_velocity(
+    """FK + inverse-dynamics first sweep for a single joint.
+
+    The world-frame child pose ``body_q[child]`` and the Newton public
+    COM-referenced twist ``body_qd[child]`` are derived from the current
+    ``joint_q`` / ``joint_qd`` while walking the articulation tree. The
+    world-frame COM transform ``X_sm`` is computed inline from
+    ``body_X_com`` instead of being read from a pre-staged ``body_q_com``
+    buffer, so the solver no longer needs a separate ``eval_fk`` or a
+    dedicated COM-staging pass (formerly a ``compute_body_q_com`` helper,
+    now removed) before the Featherstone ID kernel.
+    """
     type = joint_type[i]
```

Ruff (0.15.10)

[error] Variable `type` is shadowing a Python builtin

(A001)